sway-lib-core/src/ops.sw

```diff
@@ -173,17 +173,38 @@ impl Subtract for u64 {
     }
 }
 
-// unlike addition, underflowing subtraction does not need special treatment
-// because VM handles underflow
 impl Subtract for u32 {
     fn subtract(self, other: Self) -> Self {
-        __sub(self, other)
+        // any non-64-bit value is compiled to a u64 value under-the-hood
+        // constants (like Self::max() below) are also automatically promoted to u64
+        let res = __sub(self, other);
+        // integer underflow
+        if __gt(res, Self::max()) {
+            if panic_on_overflow_is_enabled() {
+                __revert(0)
```

Comment:
Are there any guidelines accepted on choosing revert codes in the `core`/`std`? For asserts we have specific codes and I must say they are helpful (yes, I already recognize them visually :smile:). IMO having a specif code here would also be helpful.

Reply:
It's a good point; in this case I chose to match the revert(0) used in the similar add methods:https://github.com/FuelLabs/sway/blob/0b3111771711104d4c52f6dfdf2cdc1702ea2202/sway-lib-core/src/ops.sw#L65
AFAIK there aren't any accepted guidelines for revert codes here. But this is something we should probably do as a follow up and then apply to both core/std

Reply:
For now, unless there are guidelines I'm unaware I think we should get these patches through ASAP.